docs/messaging/gcp-pubsub.md

```diff
@@ -16,9 +16,23 @@ For Google Cloud Pub/Sub, the following additional attributes are defined:
 <!-- semconv messaging.gcp_pubsub(full,tag=tech-specific-gcp-pubsub) -->
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
+| [`messaging.gcp_pubsub.message.ack_deadline`](../attributes-registry/messaging.md) | int | The ack deadline in seconds set for the modify ack deadline request. | `10` | Recommended |
+| [`messaging.gcp_pubsub.message.ack_id`](../attributes-registry/messaging.md) | string | The ack id for a given message. | `ack_id` | Recommended |
+| [`messaging.gcp_pubsub.message.delivery_attempt`](../attributes-registry/messaging.md) | int | The delivery attempt for a given message. | `2` | Recommended |
 | [`messaging.gcp_pubsub.message.ordering_key`](../attributes-registry/messaging.md) | string | The ordering key for a given message. If the attribute is not present, the message does not have an ordering key. | `ordering_key` | Conditionally Required: If the message type has an ordering key set. |
 <!-- endsemconv -->
 
+## GCP operation names
+
+In addition to the [Operation names](messaging-spans.md#operation-names), `messaging.operation` MUST be one of the following:
+
+| Value  | Description |
+|---|---|
+| `subscribe` | Streaming pull for a single message. Represents the time from after the message was received to when the message is acknowledged, negatively acknowledged, or expires. |
+| `modack` | Extends the lease for a single message or batch of messages |
+| `ack` | Acknowledges (acks) a single message or batch of messages |
```

Comment:
It sounds like different ack modes here all represent settle operation. 
With current approach in this PR, there is no way for backends to know that these operations are settling message, users would need to write custom queries for pub-sub, template dashboards would not work.

If settle operation does not work for gcp, we probably should reconsider it in the general spec.

The alternative proposal (I want to do in Azure messaging) is
- keep `messaging.operation = settle`
- have `messaging.servicebus.message.disposition_status` - this is where `complete` (ack), `abandon` (`nack`) and other statuses would go
- put specific ack mode in the span name

 i.e. span would look like
`span name="<queue> complete"`
`messaging.operation="settle"`
`messaging.servicebus.message.disposition_status="complete"`

metrics would look like
`messaging.settle.duration` and would have `messaging.servicebus.message.disposition_status` attribute.

Reply:
I agree with you ack and nack are settle operations, and I understand your point about aggregating metrics, but the GCP Pub/Sub team says they prefer having the "ack" / "nack" in the name because it corresponds with the rpc (and what users are familiar with).  I don't think they would like anything besides those values for operation. So I think we are going to leave those as is.

Alternatively, we could continue to use span_name = `my-sub ack` and set `messaging.operation` to settle? Maybe that would help with the aggregation?

I think we can generalize the status attribute, but maybe in a separate PR.

Something like `messaging.message.status`?

Reply:
> Alternatively, we could continue to use span_name = my-sub ack and set messaging.operation to settle? Maybe that would help with the aggregation?

If that works for you, that's exactly what I'm suggesting as well - `span name = "topic_name ack",  messaging.operation = settle, messaging.{?}.message.settlement_status = "ack"`. Would that work?

I like the idea of generalizing (we have at least 3 messaging systems that need it), we can take it separately.  I'd still want to add something to `messaging.message.status` like `messaging.message.settlement_status`. 

Alternatives I can think of: `messaging.message.delivery|acknowledgement|ack|consume_status`.


TBH I'd also like to entertain the idea of renaming `settle` in generic semconv to `ack|acknowledgement` - let me create a separate issue for it

Reply:
Ack, updated to set the messaging.operation attribute to the generic (settle) and we will leave the span name as the ack/nack gcp. 

For the status attributes, I agree having something generic could be good. We were thinking `messaging.gcp_pubsub.message.result` where the values are equal to ack/expire/nack. But maybe we can move this to a separate issue for when I add the streaming pull example

Reply:
I'm sorry I had to unresolve it. It seems you've updated the example, but operation names here remained unchanged.

A suggestion is to add a Span Name section above and say that settle operations would have `ack` or `nack` in  a span name. 
I added it to Azure conventions https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/azure-messaging.md, but  feel free to use any other pattern.